src/sss_client/pam_sss.c

```diff
@@ -1720,6 +1720,7 @@ static int prompt_multi_cert_gdm(pam_handle_t *pamh, struct pam_items *pi)
         ret = asprintf(&cai->choice_list_id, "%zu", c);
         if (ret == -1) {
             cai->choice_list_id = NULL;
+            free(prompt);
```

Comment:
I think `free(prompt)` is missing at the end of the loop or just before line 1714.
At the moment `cai->choice_list_id` is deallocated in line 1719 before asprintf() but the same is not true for *prompt.

Reply:
Hi,

you cannot do this because, since you are in a loop it would free the memory which was assigned in the previous cycle of the loop to `request->list.items[c++].text` in line 1729. `cai->choice_list_id` is freed because it is part of a longer living struct and might have been already set.

bye,
Sumit